lib/canary/plugs.ex

```diff
@@ -181,6 +181,27 @@ defmodule Canary.Plugs do
   plug :load_resource, model: Post, id_name: "slug", id_field: "slug", only: [:show], persisted: true
   ```
   """
+  def authorize_controller(%{assigns: %{current_user: nil}} = conn, opts) do
+    conn
+    |> Plug.Conn.assign(:authorized, false)
+    |> handle_unauthorized(opts)
+  end
+  def authorize_controller(%{assigns: %{current_user: user},
+                             private: %{phoenix_controller: controller}} = conn, opts) do
```

Comment:
I think it would be nice to allow non-phoenix users to use a controller key other than `phoenix_controller`. Maybe `canary_controller`?

So we would fetch the controller like `controller = conn.assigns[:canary_controller] || conn.assigns[:phoenix_controller]` or something like that.